public/app/features/dashboard-scene/settings/variables/VariableEditorForm.tsx

```diff
@@ -29,6 +31,9 @@ export function VariableEditorForm({ variable, onTypeChange, onGoBack, onDiscard
   const [name, setName] = React.useState(initialName ?? '');
   const [label, setLabel] = React.useState(initialLabel ?? '');
   const [description, setDescription] = React.useState(initialDescription ?? '');
```

Comment:
what is the purpose of these states?

Reply:
ah think I see how they are used. But you can just use defaultValue={...} and then only use the blur callbacks, no need for these useState

Reply:
Do not change the variable state for every `onChange`, to avoid dependencies resolving or other state changes. When onblur, we commit these changes, but in the meantime, we don't need to update the variable scene state if the value is not definitive.

Reply:
@ivanortegaalba yes, but you can do that without a React.useState, use defaultValue instead (on the inputs )

Reply:
yep, you're completely right, I did it in this way to not refactor all the old part 😅

Reply:
But why would you need to refactor the old part? If we added `defaultValue` as an optional property to the VariableTextField, VariableTextAreaField then you get rid of it straight away. The onChange/onBlur props are optional already, so should be a minimal change.

Reply:
Sorry, it was not the old part, it is the different layers on top of those components. I pushed a new commit with all the changes, now it uses the `defaultValue` 😄